tests/conftest.py

```diff
@@ -197,3 +217,50 @@ def cylc_validate_cli(capsys, caplog):
 @pytest.fixture(scope='module')
 def mod_cylc_validate_cli(mod_capsys, mod_caplog):
     return _cylc_validate_cli(mod_capsys, mod_caplog)
+
+
+@pytest.fixture
+def file_poll():
+    """Poll for the existance of a file.
+    """
+    def _inner(fpath: "Path", timeout: int = 5):
+        timeout_func(
+            fpath.exists,
+            f'file {fpath} not found after {timeout} seconds'
+        )
+    return _inner
+
+
+@pytest.fixture
+def purge_workflow(run_ok):
+    """Ensure workflow is stopped and cleaned"""
+    def _inner(id_, timeout=5):
+        run_ok(f'cylc stop {id_} --now --now')
+        clean = f'cylc clean {id_}'
+        timeout_func(
+            partial(run_ok, clean),
+            message=f'Not run after {timeout} seconds: {clean}'
+        )
+    return _inner
+
+
+@pytest.fixture
+def run_ok():
+    """Run a bash script.
+    Fail if it fails and return its output.
+    """
+    def _inner(script):
```

Comment:
VSCode/pyright does now include type info in intellisense for pytest fixtures, so would be useful to include
```suggestion
    def _inner(script: str):
```